experimental/vpx_express/custom_script.py

```diff
@@ -0,0 +1,246 @@
+                        'NS VPX is not ready to be configured, retrying in 10 seconds')
+                    time.sleep(10)
+            else:
+                logger.info('409 conflict: NS VPX already configured, no-op')
+                success = True
+                retry = False
+        except urllib2.URLError as ure:
+            logger.warning('URLError during NITRO call: reason=' +
+                           str(ure.reason))
+            if type(ure.reason).__name__ == 'timeout':
+                logger.info('Socket timeout configuring NS VPX')
+                retry_count += 1
+                if retry_count > 9:
+                    logger.info(
+                        'Too many timeouts: giving up on configuring NS VPX')
+                    retry = False
+                    failure_reason = 'Too many retries due to socket timeouts'
+                    success = False
+                    break
+                logger.info(
+                    'NS VPX is not ready to be configured, retrying in 10 seconds')
+                time.sleep(10)
+            else:
+                logger.info('Error configuring NS VPX: Irrecoverable error')
+                retry = False
+                failure_reason = 'Irrecoverable URL error'
+                success = False
+    return (success, failure_reason)
+
+
+def configure_ip(usr, pwd, ns_url, ip, subnet_mask, ip_type):
+    url = ns_url + 'nitro/v1/config/nsip'
+    logger.info('Configuring NSIP: ip= ' + ip + ', mask=' +
+                subnet_mask + ', type=' + ip_type)
+
+    jsons = '{"nsip":{"ipaddress":"%s", "netmask":"%s", "type":"%s"}}' % (
+        ip, subnet_mask, ip_type)
+    result = do_nitro_call_with_retry(
+        usr, pwd, url, jsons, timeout=10, max_retries=9)
+    if not result[0]:
+        raise Exception('Failed to configure NSIP, type=' +
+                        ip_type + ', reason=' + result[1])
+
+
+# main method
+if __name__ == "__main__":
+    logger.info("azure custom script started")
+
+    logger.info("read custom data into dict")
+    cdata = {}
+    if os.path.isfile(AZURE_CUST_DATA_FILE):
+        for pair in get_azure_custom_data():
+            k, v = get_key_value_pair(pair)
+            cdata[k] = v
+    else:
+        logger.error("failed to fetch Azure custom data")
+        exit(1)
+    logger.info("azure custom data: %s" % cdata)
+
+    logger.info("fetch networking info from metadata service")
+    attempts = 0
+    networkDict = {}
+    while True:
+        networkDict = get_azure_network_imds()
+        if networkDict:
+            break
+        elif attempts >= 6:
+            logger.error("failed to fetch network info from IMDS")
+            exit(1)
+        attempts += 1
+        time.sleep(30)
+
+    logger.info("read command line parameters")
+    if len(sys.argv) < 2:
+        logger.error("password is required for nitro calls")
+        exit(1)
+
+    try:
+        logger.info("intialize variables")
+        usr = cdata['username']
```

Comment:
If we have plan to include ssh key as part of authentication mechanism, then we need to consider different username and password. (internal user created.)